app/workers/concerns/code_analysis_mixin.rb

```diff
@@ -19,11 +31,64 @@ def merged_linter_results
     results
   end
 
+  # run linters via pronto and return the pronto result
+  def pronto_result
+    p_result = nil
+
+    # temporary solution for: download repo, obtain changes, get pronto result about changes
+    Dir.mktmpdir do |dir|
+      FileUtils.copy_entry(@branch.repo.path.to_s, dir)
+      repo = Pronto::Git::Repository.new(dir)
+      rg = repo.instance_variable_get(:@repo)
+      rg.fetch('origin', @branch.name.sub(/^prs/, 'pull'))
+      rg.checkout('FETCH_HEAD')
+      rg.reset('HEAD', :hard)
+      patches = repo.diff(@branch.merge_target)
+      p_result = Pronto::Runners.new.run(patches)
+    end
+
+    p_result
+  end
+
   def run_all_linters
-    unmerged_results = []
-    unmerged_results << Linter::Rubocop.new(branch).run
-    unmerged_results << Linter::Haml.new(branch).run
-    unmerged_results << Linter::Yaml.new(branch).run
-    unmerged_results.tap(&:compact!)
+    pronto_result.group_by(&:runner).values.map do |linted| # group by linter
+      output = {}
+      if linted.first.runner.name == "Pronto::Rubocop"
+        output["metadata"] = {
+          "rubocop_version" => ::RuboCop::Version.version,
+          "ruby_engine"     => RUBY_ENGINE,
+          "ruby_version"    => RUBY_VERSION,
+          "ruby_patchlevel" => RUBY_PATCHLEVEL.to_s,
+          "ruby_platform"   => RUBY_PLATFORM
+        }
+      end
+
+      output["files"] = linted.group_by(&:path).map do |path, value| # group by file in linter
+        {
+          "path"     => path,
+          "offenses" => value.map do |msg| # put offenses of file in linter into an array
+            {
+              "severity"  => msg.level.to_s,
+              "message"   => msg.msg,
+              "cop_name"  => msg.runner,
+              "corrected" => false,
+              "location"  => {
+                "line"   => msg.line.position,
+                "column" => 0, # TODO: value cannot be obtained from Pronto::Message
+                "length" => msg.line.length
+              }
+            }
+          end
+        }
+      end
+
+      output["summary"] = {
+        "offense_count"        => output["files"].reduce(0) { |sum, item| sum + item['offenses'].length },
+        "target_file_count"    => output["files"].length,
+        "inspected_file_count" => 0 # TODO: value cannot be obtained from the result of `pronto_result` method
```

Comment:
Do we even use this in the output comment?

Reply:
@Fryguy it is not used, it was used only in /lib/linters, the output comment does not use it https://github.com/ManageIQ/miq_bot/blob/master/app/workers/commit_monitor_handlers/commit_range/rubocop_checker/message_builder.rb#L44
I will remove it.